js/src/components/tree-select-control/options.js

```diff
@@ -81,17 +84,39 @@ const Options = ( {
 	 * @param {Option} option The option to be expanded or collapsed
 	 */
 	const toggleExpanded = ( option ) => {
+		if ( ! option.children?.length ) return;
+
 		onNodesExpandedChange(
 			nodesExpanded.includes( option.value )
 				? nodesExpanded.filter( ( el ) => option.value !== el )
 				: [ ...nodesExpanded, option.value ]
 		);
 	};
 
-	return options.map( ( option ) => {
+	/**
+	 * Alters the node with some keys for accessibility
+	 * ArrowRight - Expands the node
+	 * ArrowLeft - Collapses the node
+	 * Space - Checks/Unchecks the node
+	 *
+	 * @param {Event} event The KeyDown event
+	 * @param {Option} option The option where the event happened
+	 * @param {boolean} isExpanded True if the node is expanded, false otherwise
+	 */
+	const handleKeyDown = ( event, option, isExpanded ) => {
+		if ( event.key === ARROW_RIGHT && ! isExpanded ) {
+			toggleExpanded( option );
+		} else if ( event.key === ARROW_LEFT && isExpanded ) {
+			toggleExpanded( option );
+		} else if ( event.key === SPACE ) {
```

Comment:
❔ According to the spec, looks like the toggling is also supported by the Enter key.

Reply:
Not sure how accurate are the specs with the real needs. Since we are selecting with SPACE the node and we don't have any Key for expanding the tree itself when the user closes it with ESC I thought it will be more useful ENTER for that.

Reply:
This makes sense, thanks for the explanation!